### ISSUE TITLE
[Bug]: Error upon plan/apply of `aws_appautoscaling_target` on v4.66.0

### ISSUE THREAD

**Issue by darkwizard242** [hashicorp/terraform-provider-aws#31180]
### Terraform Core Version

0.12.31

### AWS Provider Version

4.66.0

### Affected Resource(s)

aws_appautoscaling_target

### Expected Behavior

Resource provisioning with or without tags.

### Actual Behavior

Results in an error snippet when tried with or without tags for `aws_appautoscaling_target` resource. Please note that the same code was working with v4.65.0 of the Terraform AWS provider and the mentioned Terraform version.

Change introduced in [v4.66.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v4.66.0). 

### Relevant Error/Panic Output Snippet

```shell
Error: listing tags for Application Auto Scaling Target (): InvalidParameter: 1 validation error(s) found.
minimum field size of 1, ListTagsForResourceInput.ResourceARN.
```

### Terraform Configuration Files
```hcl
resource "aws_appautoscaling_target" "ecs_target" {
  max_capacity       = 4
  min_capacity       = 1
  resource_id        = "service/${aws_ecs_cluster.example.name}/${aws_ecs_service.example.name}"
  scalable_dimension = "ecs:service:DesiredCount"
  service_namespace  = "ecs"
}
```

### Steps to Reproduce
`terrform plan`
OR
`terraform apply`

### Would you like to implement a fix?

No

**Comment by jordanb-afs** [comment]
i'm also seeing this after 4.66 got picked up

**Comment by xshot9011** [comment]
I have temporarily restricted the version of the provider within my module.

```terraform
terraform {
  required_version = ">= 1.0.0"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "<= 4.65"
    }
    random = {
      source  = "hashicorp/random"
      version = ">= 2.3.0"
    }
  }
}

```

**Comment by cgarcia-l** [comment]
I have the same problem with AWS Provider Version 4.66.0 and Terraform Core Version 1.4.6

**Comment by jeremychauvet** [comment]
Relates #30172

**Comment by dgokcin** [comment]
we have the same problem. Any estimates for a fix?

**Comment by monaboiste** [comment]
we're also experiencing the same problem, but only with the provisioned capacity provider. The fargate spot works fine (Terraform v0.14.6)

**Comment by manav0191** [comment]
Hello , 
I was facing the same problem, this is specifically related to version 4.66.0 meanwhile,
I have fixed it by hardcoding the version to "4.65.0"
If you have different version in the locked provider just use

"terraform init upgrade"  (To match the version as your config)


```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.65.0" 
}
}
}
```

**Comment by balonik** [comment]
Same issue with our setup

```
resource "aws_appautoscaling_target" "dev_msk_storage_scaling_target" {
  max_capacity       = 2000
  min_capacity       = 1
  resource_id        = aws_msk_cluster.dev.arn
  scalable_dimension = "kafka:broker-storage:VolumeSize"
  service_namespace  = "kafka"
}
```

**Comment by ewbankkit** [comment]
It looks like the problem is that some (all?) targets created before 2023-03-20 (when the tagging feature was added to the AWS SDK) don't have an ARN assigned. The ARN is used to make the `ListTags` call which is happening on `terraform refresh`.
We'll fix this and get a patch release published.

**Comment by upasanaeverestek** [comment]
Same issue with my setup also, I am using 1.4.2 terraform version

**Comment by YLgearbox** [comment]
I found the same bug with auto scaling, looking forward to a newer provider version to fix it.

**Comment by ricardobf** [comment]
I'm facing the same issue,
aws version `4.38`.

**Comment by vadym-sedna** [comment]
@ewbankkit when we expect the fix to be available? still failing on our side. thx!

**Comment by mathieuforest** [comment]
@vadym-sedna the pipeline is running https://github.com/hashicorp/terraform-provider-aws

**Comment by ewbankkit** [comment]
The fix is available in **v4.66.1** of the Terraform AWS Provider: https://registry.terraform.io/providers/hashicorp/aws/4.66.1.
